hbase-server/src/main/java/org/apache/hadoop/hbase/master/AssignmentManager.java

```diff
@@ -3343,6 +3346,55 @@ boolean waitUntilNoRegionsInTransition(final long timeout)
     return offlineServers;
   }
 
+  void deleteNonZkBasedQualifiersForZkBasedAssignment() throws IOException {
+    boolean isZKAssignmentInUse =
+      ConfigUtil.useZKForAssignment(server.getConfiguration()) && !server.getConfiguration()
+        .getBoolean("hbase.assignment.usezk.migrating", false);
+    if (isZKAssignmentInUse) {
+      List<Result> results = MetaTableAccessor.fullScanOfMeta(server.getConnection());
+      List<Delete> redundantCQDeletes = new ArrayList<>();
+      for (Result result : results) {
+        RegionLocations rl =  MetaTableAccessor.getRegionLocations(result);
+        if (rl == null) {
+          continue;
+        }
+        HRegionLocation[] locations = rl.getRegionLocations();
+        if (locations == null) {
+          continue;
+        }
+        for (HRegionLocation hrl : locations) {
+          if (hrl == null) {
+            continue;
+          }
+          HRegionInfo regionInfo = hrl.getRegionInfo();
+          if (regionInfo == null) {
+            continue;
+          }
+          int replicaId = regionInfo.getReplicaId();
+          Cell cell = result.getColumnLatestCell(HConstants.CATALOG_FAMILY,
+            RegionStateStore.getServerNameColumn(replicaId));
+          if (cell != null && cell.getValueLength() > 0) {
+            Delete delete =
+              new Delete(cell.getRowArray(), cell.getRowOffset(), cell.getRowLength());
+            delete.addColumns(HConstants.CATALOG_FAMILY, HConstants.SERVERNAME_QUALIFIER);
```

Comment:
is this sufficient for replicas? Doesn't it need to be
```suggestion
            delete.addColumns(HConstants.CATALOG_FAMILY, RegionStateStore.getServerNameColumn(replicaId));
```